cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot.go

```diff
@@ -20,27 +20,26 @@ import (
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/autoscaler/cluster-autoscaler/simulator/clustersnapshot"
 	"k8s.io/autoscaler/cluster-autoscaler/simulator/framework"
-	"k8s.io/autoscaler/cluster-autoscaler/simulator/predicatechecker"
 )
 
 // PredicateSnapshot implements ClusterSnapshot on top of a SnapshotBase by using
 // SchedulerBasedPredicateChecker to check scheduler predicates.
 type PredicateSnapshot struct {
 	clustersnapshot.SnapshotBase
-	predicateChecker *predicatechecker.SchedulerBasedPredicateChecker
+	pluginRunner *SchedulerPluginRunner
```

Comment:
nit: Maybe this could be using some abstract `PluginRunner` interface? Would make it easier to run unit tests.

Reply:
Ack